### PR TITLE
Login using Token or OAuth

### PR DESCRIPTION
**Pull request by spahrson**

### CODE REVIEW

File: src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java [hunk 1]
```diff
@@ -51,16 +67,55 @@
     protected final CompletableFuture<String> login;
     protected final Executor executor;
 
+    @Deprecated
     public RocketChatClient(String url, String user, String password) {
         this(url, user, password, ForkJoinPool.commonPool());
     }
 
+    @Deprecated
     public RocketChatClient(String url, String user, String password, Executor executor) {
```
Comment (daniel-sc) [inline src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java:74]: Would it be cleaner to invoke the new constructor: `this(url, new LoginParam(user, password), executor)`?

File: src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java [hunk 2]
```diff
@@ -51,16 +67,55 @@
     protected final CompletableFuture<String> login;
     protected final Executor executor;
 
+    @Deprecated
```
Comment (daniel-sc) [inline src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java]: Is there a specific reason for deprecating the old constructors?

File: src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java [hunk 3]
```diff
@@ -76,18 +131,30 @@ public RocketChatClient(String url, String user, String password, Executor execu
...
 
-    protected CompletableFuture<String> login(String user, String password) {
-        return connect().thenComposeAsync(session -> sendDirect(new MethodRequest("login", new LoginParam(user, password)),
-                failOnError(r -> r.result.getAsJsonObject().get("token").getAsString())), executor);
+    protected CompletableFuture<String> login(LoginParam param) {
+        return connect().thenComposeAsync(session -> sendDirect(new MethodRequest("login", param),
+            failOnError(r -> r.result.getAsJsonObject().get("token").getAsString())), executor);
+    }
+
+    private CompletableFuture<String> loginWithToken(String token) {
```
Comment (daniel-sc) [inline src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java]: Would it make sense to create a class `LoginTokenParam` analogous `LoginOAuthParam` and `LoginParam`?

**Review by daniel-sc — CHANGES_REQUESTED** [review]
Hi @spahrson,
thanks for your contribution! 
After addressing the comments (via update or answer) I'm happy to merge this!

File: src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java (hunk 1 above)
Comment (spahrson) [reply src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java:74]: > Would it be cleaner to invoke the new constructor: `this(url, new LoginParam(user, password), executor)`?

my fault. i'm going to fix this.

File: src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java (hunk 2 above)
Comment (spahrson) [reply src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java]: I added a new constructor taking the loginParam. So there are now two ways to achive the same. IMHO we should eliminate one of them to keep it simple and clean?

File: src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java (hunk 3 above)
Comment (spahrson) [reply src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java]: Changed.

File: src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java (hunk 2 above)
Comment (daniel-sc) [reply src/main/java/com/github/daniel_sc/rocketchat/modern_client/RocketChatClient.java]: If rather keep the simple additional interface. But if you still think deprecation is better, that’s fine with me, as long as you add some javadoc hinting to the new constructor.

**Review by daniel-sc — APPROVED** [review]
@spahrson Thanks for the updates!

**Pull request merged**